src/wrappers.jl

```diff
@@ -0,0 +1,442 @@
+
+```juila
+using GeoInterface.Wrappers
+```
+
+Otherwise they can be accessed as:
+
+```julia
+GeoInterface.Polygon
+```
+"""
+module Wrappers
+
+import Extents
+
+import ..GeoInterface: AbstractGeometryTrait, PointTrait, LineTrait, LineStringTrait, LinearRingTrait,
+       MultiPointTrait, PolygonTrait, MultiLineStringTrait, MultiCurveTrait, MultiPolygonTrait,
+       AbstractCurveTrait, GeometryCollectionTrait, FeatureTrait, FeatureCollectionTrait, PointTuple3,
+       PolyhedralSurfaceTrait, TriangleTrait, QuadTrait, PentagonTrait, HexagonTrait, RectangleTrait, TINTrait
+
+import ..GeoInterface: isgeometry, isfeature, isfeaturecollection, is3d, ismeasured,
+       trait, geomtrait, convert, x, y, z, m, extent, crs,
+       getgeom, getpoint, getring, gethole, getcoord,
+       ngeom, npoint, nring, nhole, ncoord,
+       nfeature, getfeature, geometry, properties
+
+export Point, Line, LineString, LinearRing, Polygon, MultiPoint, MultiPolygon, MultiLineString, MultiCurve,
+    PolyhedralSurface, GeometryCollection, Feature, FeatureCollection
+
+# TODO
+# Triangle, Quad, Pentagon, Hexagon, TIN, Surface
+
+
+# Implement interface
+
+"""
+    abstract type WrapperGeometry{Z,M,T}
+
+Provides geometry wrappers that wrap any GeoInterface compatible
+objects, or vectors or their child objects.
+
+These can be useful for building custom geometries, in tests,
+and in packages with no direct dependencies on specific geometry
+types.
+
+Parameters `Z` and `M` hold `Bool` values `true` or `false`
+to indicate if a z dimension or measures are present. They are
+usually detected from the wrapped object, but can be added manually
+e.g. for `Tuple` or `Vector` points to have the third value used as
+measures.
+"""
+abstract type WrapperGeometry{Z,M,T} end
+
+isgeometry(::Type{<:WrapperGeometry}) = true
+is3d(::WrapperGeometry{Z}) where Z = Z
+ismeasured(::WrapperGeometry{<:Any,M})  where M = M
+
+Base.parent(geom::WrapperGeometry) = geom.geom
+
+function Base.:(==)(g1::WrapperGeometry, g2::WrapperGeometry)
+    all(((a, b),) -> a == b, zip(getgeom(g1), getgeom(g2)))
+end
+
+geointerface_geomtype(trait) = throw(ArgumentError("trait $trait not yet implemented as a GeoInterface geometry wrapper"))
+
+# Interface methods
+# With indexing
+function getgeom(trait::AbstractGeometryTrait, geom::WrapperGeometry{<:Any,<:Any,T}, i) where T
+    isgeometry(T) ? getgeom(trait, parent(geom), i) : parent(geom)[i]
+end
+function getgeom(trait::AbstractGeometryTrait, geom::WrapperGeometry{<:Any,<:Any,T}) where T
+    isgeometry(T) ? getgeom(trait, parent(geom)) : parent(geom)
+end
+
+extent(::AbstractGeometryTrait, geom::WrapperGeometry) =
+    isnothing(geom.extent) && isgeometry(parent(geom)) ? extent(parent(geom)) : geom.extent
+function ngeom(trait::AbstractGeometryTrait, geom::WrapperGeometry{<:Any,<:Any,T}) where T
+    isgeometry(T) ? ngeom(parent(geom)) : length(parent(geom))
+end
+
+for (geomtype, trait, childtype, child_trait, length_check, nesting) in (
```

Comment:
Might be good to explain length_check and nesting for future additions to this list.